app/controllers/discourse_assign/assign_controller.rb

```diff
@@ -114,8 +115,64 @@ def assigned
       render json: { topics: serialize_data(topics, AssignedTopicSerializer) }
     end
 
+    def group_assignments
```

Comment:
A lot of this code seems similar to `def assigned` above. Can we share any logic between them?